zkml/src/parser/onnx.rs

```diff
@@ -308,51 +318,50 @@ fn load_gemm(
     // must make sure one of the inputs is the current matrix node. Otherwise that's just a normal add that we don't support.
     // TODO: support general case with Add layer.
     let mut piter = iter.peekable();
-    let next_node_id = **piter
-        .peek()
-        .ok_or(ProvableOpError::OnnxParsingError(format!(
-            "Gemm {} has no next node",
-            node.name
-        )))?;
-
-    let next_node = model.node(next_node_id);
-    // if there's a bias, the next op is a TypedBinOp( Add ) node
-    let bias_node_id = match downcast_to::<TypedBinOp>(next_node) {
-        // safety net
-        _ if next_node.inputs.len() != 2 => {
-            // no bias, just return the matrix node
-            None
-        }
-        // the operation must be an Add
-        Ok(binop) if binop.0.is::<tract_core::ops::math::Add>() => {
-            // now on this node, we need to ensure one of the inputs is the current matrix node
-            match next_node
-                .inputs
-                .iter()
-                .enumerate()
-                .find(|(_i, &x)| x.node == node_id)
-            {
-                Some((idx, ..)) => {
-                    // Now we need to find the bias node, which is the other input to the Add node
-                    // and we can extract it as a constant tensor afterwards
-                    // since only two elements, we can just do 1 - idx
-                    let bias_input = next_node.inputs[1 - idx];
-                    // let bias_node = model.node(bias_input.node);
-                    // in that case, we move on the iterator, since we already saw the bias node and the Add is part of the dense layer
-                    // unwrap is safe here since we peeked already
-                    piter.next().unwrap();
-                    Some(bias_input.node)
+    let (edge_id, bias_node_id) = match piter.peek() {
+        // no next node, no bias
+        None => (node_id, None),
+        Some(&&next_node_id) => (next_node_id, {
```

Comment:
Shouldn't we return `node_id` instead of `next_node_id` in cases where `bias_node_id` is `None`? As the other layers will refer to the `Einsum` node id if there is no bias, as it was before these changes afaiu?